scapy/contrib/gmlan.uts

```diff
@@ -0,0 +1,342 @@
+# gmlan unit tests
+#
+# Type the following command to launch start the tests:
+# $ sudo bash test/run_tests -t test/gmlan.uts -F
+
+% gmlan unit tests
+
++ Configuration of scapy
+= Load gmlan layer
+~ conf command
+
+load_contrib('gmlan')
+
++ Basic Packet Tests()
+= Set GMLAN ECU AddressingScheme
+conf.contribs['GMLAN']['GMLAN_ECU_AddressingScheme'] = 2
+conf.contribs['GMLAN']['GMLAN_ECU_AddressingScheme'] == 2
+
+= Craft Packet
+x = GMLAN(b'\x52\x02\x01\x16\x71\x00\x00\x0c\xaa\xbb')
+x.PIDDataBytes == b'\xaa\xbb'
+x.service == 0x52
+
+= Craft VIN Packet
+x = GMLAN(b'\x5a\x90'+bytes("WOOOJBF35W1042000",'utf-8'))
```

Comment:
You can use the `raw()` function if needed (cross-python)

Reply:
Actually `bytes()` also works with Python 2.7. Let's just say it's more-Scapy ;-)

Reply:
Done